backend-rust/src/indexer.rs

```diff
@@ -335,96 +335,112 @@ impl Indexer for BlockPreProcessor {
+                    let items = client4
+                        .get_block_items(fbi.height)
+                        .await?
+                        .response
+                        .try_collect::<Vec<_>>()
+                        .await?;
+                    Ok(items)
+                };
 
-            let get_special_items = async move {
-                let items = client5
-                    .get_block_special_events(fbi.height)
-                    .await?
-                    .response
-                    .try_collect::<Vec<_>>()
-                    .await?;
-                Ok(items)
-            };
+                let get_special_items = async move {
+                    let items = client5
+                        .get_block_special_events(fbi.height)
+                        .await?
+                        .response
+                        .try_collect::<Vec<_>>()
+                        .await?;
+                    Ok(items)
+                };
+                let start_fetching = Instant::now();
+                let (block_info, chain_parameters, events, items, tokenomics_info, special_events) =
+                    try_join!(
+                        client1.get_block_info(fbi.height),
+                        client2.get_block_chain_parameters(fbi.height),
+                        get_events,
+                        get_items,
+                        client.get_tokenomics_info(fbi.height),
+                        get_special_items
+                    )?;
+                let get_total_staked_capital = async move {
+                    match tokenomics_info.response {
+                        RewardsOverview::V0 {
+                            ..
+                        } => {
+                            compute_total_stake_capital(
+                                &mut client6,
+                                BlockIdentifier::AbsoluteHeight(fbi.height),
+                            )
+                            .await
+                        }
+                        RewardsOverview::V1 {
+                            total_staked_capital,
+                            ..
+                        } => Ok(total_staked_capital),
+                    }
+                };
+                let get_block_certificates = async move {
+                    // Fetching the block certificates prior to P6 results in a NotFound gRPC error,
+                    // so we produce the empty type of certificates instead.
+                    if block_info.response.protocol_version < ProtocolVersion::P6 {
+                        Ok(BlockCertificates {
+                            quorum_certificate:       None,
+                            timeout_certificate:      None,
+                            epoch_finalization_entry: None,
+                        })
+                    } else {
+                        let response = client7.get_block_certificates(fbi.height).await?;
+                        Ok(response.response)
```

Comment:
Currently, we only use the fetched data for ProtocolVersion::P8 and above. If it stays like this we could avoid fetching the data here for protocolVersions below (minor optimization).